onnx/reference/ops/op_if.py

```diff
@@ -65,4 +65,4 @@ def _run(
                     f"Output {i!r} (branch={branch!r}, name={names[i]!r}) is None, "
                     f"available inputs={sorted(context)}, initializers={inits}."
                 )
-        return final
+        return self._check_output_run(final)
```

Comment:
Was there a reason we need to call this outside of the base class? When do we need to call it in implementations?

Reply:
It should not be called. The implementation should overwrite method `_run` and not `run`.

Reply:
Does it mean it should be removed from the implementations?

Reply:
I mean, `run` calls `_run` and `_check_output_run` for the user. For a new op, the user should only overwrite `_run`.